src/framework/workspace/internal/workspace.cpp

```diff
@@ -32,81 +32,145 @@ using namespace muse;
 using namespace muse::workspace;
 
 Workspace::Workspace(const io::path_t& filePath, const modularity::ContextPtr& iocCtx)
-    : Injectable(iocCtx), m_file(filePath)
+    : Injectable(iocCtx)
 {
+    m_file = std::make_shared<WorkspaceFile>(filePath);
+
     multiInstancesProvider()->resourceChanged().onReceive(this, [this](const std::string& resourceName){
         if (resourceName == fileResourceName()) {
             reload();
         }
     });
 }
 
+Workspace::Workspace(const io::path_t& filePath, const Workspace* other, const ContextPtr& iocCtx)
+    : Workspace(filePath, iocCtx)
+{
+    m_file = std::make_shared<WorkspaceFile>(filePath, other->m_file.get());
+}
+
 std::string Workspace::name() const
 {
-    return io::completeBasename(m_file.filePath()).toStdString();
+    return io::completeBasename(m_file->filePath()).toStdString();
 }
 
-std::string Workspace::title() const
+void Workspace::setName(const std::string& name)
```

Comment:
I would like a different name here, now it looks like just a setter, but it does a lot of things